src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,217 @@
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WebRTCTransportProvider {
+
+using ICEServerStruct     = Structs::ICEServerStruct::DecodableType;
+using WebRTCSessionStruct = Structs::WebRTCSessionStruct::Type;
+
+/**
+ * @brief
+ *   A delegate interface for the WebRTC Transport Provider cluster, allowing
+ *   OEM or application-specific logic to be implemented outside of the cluster
+ *   server itself. The server calls these methods to do the "real work."
+ */
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    /**
+     * @brief
+     *   This method is called when the server receives the SolicitOffer command.
+     *
+     *   It is used by the requestor to ask the provider to initiate a new Offer/Answer
+     *   negotiation. The specification states that the provider may be in a standby mode
+     *   (low power), thus a "deferred offer" scenario might occur if the device needs time
+     *   to fully power resources needed for streaming.
+     *
+     * @param[in]  id               A local ID for the session being allocated
+     * @param[in]  streamUsage      Indicates the usage for this session
+     * @param[in]  videoStreamId    The requested VideoStreamID:
+     *                                - Not present => No video requested
+     *                                - Null => Automatic assignment requested
+     *                                - Valid => Use the specified ID
+     * @param[in]  audioStreamId    The requested AudioStreamID:
+     *                                - Not present => No audio requested
+     *                                - Null => Automatic assignment requested
+     *                                - Valid => Use the specified ID
+     * @param[in]  iceServers       Optional list of ICE server configurations (TURN/STUN).
+     * @param[in]  iceTransportPolicy  If present, indicates whether ICE transport is 'all' or 'relay' only.
+     * @param[in]  metadataOptions  Reserved for future use (value = 0).
+     * @param[in]  peerNodeId       The Node ID of the controller or requestor.
+     * @param[in]  peerFabricIndex  The FabricIndex of the requestor.
+     * @param[out] outSession       On success, must be populated with the new session's details.
+     * @param[out] outDeferredOffer True if the device is in standby or needs to defer generating an Offer.
+     *
+     * @return CHIP_ERROR
+     *   - CHIP_NO_ERROR if a session was successfully created or prepared for negotiation
+     *   - Appropriate error otherwise (e.g. out of resources)
+     */
+    virtual CHIP_ERROR HandleSolicitOffer(uint16_t id, StreamUsageEnum streamUsage,
```

Comment:
This has 11 parameters with many having identical types. Could we create an args type instead please? then the names will be clearer to read